### PR TITLE
Implement `$bitsAllClear`

### PR DESCRIPTION
**Pull request by w84thesun**
Refs #256.

### CODE REVIEW

**Comment by AlekSi** [comment]
That PR handles only `$bitsAllClear`, right? Why it closes #256?

File: internal/handlers/common/params.go [hunk 1]
```diff
@@ -93,3 +93,89 @@ func GetNumberParam(parameterName string, value any) (int64, error) {
 	}
 	return numberValue, nil
 }
+
+// GetBinaryMaskParam returns *types.Binary value matched and error if match failed.
+// Possible values are: position array ([1,3,5] == 010101), integer value and types.Binary value.
+func GetBinaryMaskParam(value any) (mask *types.Binary, err error) {
```
Comment (AlekSi) [inline internal/handlers/common/params.go:99]: Let's return types.Binary by value as we do with all scalar types (even if that's a small structure).

Let's also do not use named results if we don't actually use them

File: internal/handlers/common/params.go [hunk 2]
```diff
@@ -93,3 +93,95 @@ func GetNumberParam(parameterName string, value any) (int64, error) {
...
+		return types.Binary{}, NewErrorMsg(ErrBadValue,
+			fmt.Sprintf(
+				`value takes an Array, a number, or a BinData but received: $bitsAllClear: %#v`, value),
+		)
+	}
+	return mask, nil
+}
+
+// GetBinaryParam matches value type, returning types.Binary and error if match failed.
+func GetBinaryParam(value any) (types.Binary, error) {
```
Comment (AlekSi) [inline internal/handlers/common/params.go:159]: Seems unused

File: internal/types/binary_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBinaryFromArray(t *testing.T) {
```
Comment (AlekSi) [inline internal/types/binary_test.go:23]: The linter is correct there

File: internal/handlers/common/params.go (hunk 2 above)
Comment (w84thesun) [reply internal/handlers/common/params.go]: It's used in `filterFieldExprBitsAllClear`

Comment (AlekSi) [reply internal/handlers/common/params.go]: Please not resolve the conversation if you want your answer to be seen :)

File: internal/types/binary_test.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		{
+			name: "bits 1,3,5",
+			args: args{values: MustNewArray(int32(1), int32(3), int32(5))},
+			want: Binary{
+				Subtype: BinaryGeneric,
+				B:       []byte{0b101010, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
+			},
+		},
+	}
+	for _, tt := range tests {
```
Comment (AlekSi) [inline internal/types/binary_test.go:55]: There is another good point by linter

Comment (w84thesun) [reply internal/types/binary_test.go:55]: Fixed

**Review by AlekSi — DISMISSED** [review]

**Comment by AlekSi** [comment]
@w84thesun why you self-assigned that PR?

**Review by AlekSi — APPROVED** [review]

**Pull request merged**